src/pudl/output/ferc1.py

```diff
@@ -422,6 +445,53 @@ def out_ferc1__yearly_steam_plants_fuel_sched402(
     return fuel_df
 
 
+## example/draft factory pattern
+specs = [
+    {"name": "ferc1__yearly_purchased_power_and_exchanges_sched326"},
+    {"name": "ferc1__yearly_plant_in_service_sched204"},
+    {"name": "ferc1__yearly_balance_sheet_assets_sched110"},
+]
+
+
+# draft asset_factory
+def generate_asset_factory(spec) -> AssetsDefinition:
+    var_name = "core_" + spec["name"]
+    core_ = globals()[var_name]
+
+    @asset(
+        name=f"_out_{get_core_ferc1_asset_description(spec["name"])}",
+        io_manager_key="pudl_io_manager",
+        compute_kind="Python",
+    )
+    def _asset(
+        core_: pd.DataFrame,
+        core_pudl__assn_ferc1_pudl_utilities: pd.DataFrame,
+    ) -> pd.DataFrame:
+        """Generate a dataframe for {} asset specification.""".format(spec["name"])
+        return_df = core_.merge(
+            core_pudl__assn_ferc1_pudl_utilities, on="utility_id_ferc1"
+        )
+        return return_df
+
+    return _asset
```

Comment:
A more specific name would be helpful, and I think this can be as simple as just taking a string as input:

```suggestion
def ferc1_output_asset_factory(table_name: str) -> AssetsDefinition:
    @asset(
        name=f"out_ferc1__{table_name}",
        ins={
            core_pudl__assn_ferc1_pudl_utilities: AssetIn(),
            f"core_ferc1__{table_name}": AssetIn(),
        },
        io_manager_key="pudl_io_manager",
        compute_kind="Python",
    )
    def _ferc1_output_asset(**kwargs) -> pd.DataFrame:
        f"""Generate a dataframe for out_ferc1__{table_name} asset specification."""
        return kwargs[core_ferc1__{table_name}].merge(
            core_pudl__assn_ferc1_pudl_utilities, on="utility_id_ferc1"
        )

    return _ferc1_output_asset
```